### PR TITLE
feat: Table caption

### LINKED ISSUE #43036: Table title prop deprecated
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/stykts?file=/demo.tsx)

### Steps to reproduce

title prop is deprecated — Use caption instead

### What is expected?

no error/warning shown

### What is actually happening?

title prop is working but with a warning about deprecation

| Environment | Info |
| --- | --- |
| antd | 5.6.1 |
| React | "react": "^18.2.0" |
| System | MacOS and CodeSandbox |
| Browser | Chrome 108.0.5359.124 |

### PR DESCRIPTION
**Pull request by sarpere**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
close #43036

### 💡 Background and solution

Regarding to rc-table component, title prop is deprecated. I replaced title prop with caption prop. 

Screenshot 2023-07-06 at 12 32 51 am

### 📝 Changelog

Table title prop replaced with caption prop.

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    -       |
| 🇨🇳 Chinese |      -     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at a070e15</samp>

This pull request enhances the Table component to use the `caption` prop instead of the `title` prop for table captions, following the HTML standard and improving accessibility. It also updates the documentation and demos in both English and Chinese to reflect this change.

### <samp>🤖 Generated by Copilot at a070e15</samp>

* Replace the title prop with the caption prop in the Table component API and the bordered table demo, to support HTML captions for accessibility and consistency ([link](https://github.com/ant-design/ant-design/pull/43172/files?diff=unified&w=0#diff-5a56dc8f52d7cfd77eccbfc352bd52ac0a23e24ffbcc71ad548b2fe381e7d69eL56-R56), [link](https://github.com/ant-design/ant-design/pull/43172/files?diff=unified&w=0#diff-8d3a4194d516aa3322b22843bc2a379cdb0ad134c5e22899e6468d16c85d85c9L136-R136), [link](https://github.com/ant-design/ant-design/pull/43172/files?diff=unified&w=0#diff-1d555de1f30d5e46ff95e81b3d10b27baf15321e455557e7d3a55ca09ff61209L137-R137), [link](https://github.com/ant-design/ant-design/pull/43172/files?diff=unified&w=0#diff-8d3a4194d516aa3322b22843bc2a379cdb0ad134c5e22899e6468d16c85d85c9L81-R81), [link](https://github.com/ant-design/ant-design/pull/43172/files?diff=unified&w=0#diff-6e1f33c8981440ca04ca723f36c435558a74e0119edd5934edae16ebce591104L7-R7))
* Style the caption element with some padding and left alignment in the bordered table demo (`bordered.tsx`) ([link](https://github.com/ant-design/ant-design/pull/43172/files?diff=unified&w=0#diff-5a56dc8f52d7cfd77eccbfc352bd52ac0a23e24ffbcc71ad548b2fe381e7d69eL56-R56))
* Change the import order of the bordered table demo (`bordered.tsx`) to follow the convention of importing React first, then antd components, then types ([link](https://github.com/ant-design/ant-design/pull/43172/files?diff=unified&w=0#diff-5a56dc8f52d7cfd77eccbfc352bd52ac0a23e24ffbcc71ad548b2fe381e7d69eL1-R3))

### CODE REVIEW

File: components/table/index.en-US.md [hunk 1]
```diff
@@ -133,7 +133,7 @@ const columns = [
 | sticky | Set sticky header and scroll bar | boolean \| `{offsetHeader?: number, offsetScroll?: number, getContainer?: () => HTMLElement}` | - | 4.6.0 (getContainer: 4.7.0) |
 | summary | Summary content | (currentData) => ReactNode | - |  |
 | tableLayout | The [table-layout](https://developer.mozilla.org/en-US/docs/Web/CSS/table-layout) attribute of table element | - \| `auto` \| `fixed` | -<hr />`fixed` when header/columns are fixed, or using `column.ellipsis` |  |
-| title | Table title renderer | function(currentPageData) | - |  |
+| caption | Set caption of Table | ReactNode | - |  |
```
Comment (afc163) [inline components/table/index.en-US.md:136]:
```suggestion
| caption | Set caption of Table, try deprecated prop `title` before 5.7.0 | ReactNode | - | 5.7.0  |
```

File: components/table/index.zh-CN.md [hunk 2]
```diff
@@ -134,7 +134,7 @@ const columns = [
 | sticky | 设置粘性头部和滚动条 | boolean \| `{offsetHeader?: number, offsetScroll?: number, getContainer?: () => HTMLElement}` | - | 4.6.0 (getContainer: 4.7.0) |
 | summary | 总结栏 | (currentData) => ReactNode | - |  |
 | tableLayout | 表格元素的 [table-layout](https://developer.mozilla.org/zh-CN/docs/Web/CSS/table-layout) 属性，设为 `fixed` 表示内容不会影响列的布局 | - \| `auto` \| `fixed` | 无<hr />固定表头/列或使用了 `column.ellipsis` 时，默认值为 `fixed` |  |
-| title | 表格标题 | function(currentPageData) | - |  |
+| caption | 表格标题 | ReactNode | - |  |
```
Comment (afc163) [inline components/table/index.zh-CN.md:137]:
```suggestion
| caption | 表格标题，在 5.7.0 之前使用 `title` 属性 | ReactNode | - | 5.7.0  |
```

File: components/table/demo/bordered.tsx [hunk 3]
```diff
@@ -53,7 +53,7 @@ const App: React.FC = () => (
     columns={columns}
     dataSource={data}
     bordered
-    title={() => 'Header'}
+    caption={<div style={{ textAlign: 'left', padding: 16 }}>Header</div>}
```
Comment (afc163) [inline components/table/demo/bordered.tsx:56]:
```
title={() => 'Header'}
```

and

```
caption={() => 'Header'}
```

should have same styles by default, which it is not now.

图片

**Review by afc163 — CHANGES_REQUESTED** [review]

File: components/table/demo/bordered.tsx (hunk 3 above)
Comment (sarpere) [reply components/table/demo/bordered.tsx:56]: I think, it is an issue related to rc-table. Do you want me to create a workaround from our project? Or, I can commit an bug fix to `rc-table`.

Comment (afc163) [reply components/table/demo/bordered.tsx:56]: https://github.com/ant-design/ant-design/blob/master/components/table/style/index.ts#L131

https://github.com/ant-design/ant-design/blob/master/components/table/style/bordered.ts#L34-L37

The default styles for Table `title` is here. We could add styles for caption.

Comment (sarpere) [reply components/table/demo/bordered.tsx:56]: Got it. Let me fix the style issue.

**Comment by sarpere** [comment]
I noticed that there is a issue on caption missing right border. Same [issue](https://github.com/react-component/table/issues/999) exist in `rc-table`. 

Screenshot 2023-06-25 at 11 45 09 pm

File: components/table/__tests__/__snapshots__/demo-extend.test.ts.snap [hunk 4]
```diff
@@ -23619,6 +23615,8 @@ Array [
         aria-valuenow="50"
         autocomplete="off"
         class="ant-input-number-input"
+        classes="[object Object]"
+        classnames="[object Object]"
```
Comment (afc163) [inline components/table/__tests__/__snapshots__/demo-extend.test.ts.snap:23619]: 这是啥？

Comment (sarpere) [reply components/table/__tests__/__snapshots__/demo-extend.test.ts.snap:23619]: I accidentally merged feature branch. I suppose that It came from a different branch.

Comment (sarpere) [reply components/table/__tests__/__snapshots__/demo-extend.test.ts.snap:23619]: Do you want me to make any changes on the PR? 
@afc163

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/actions/runs/5411720569/jobs/9835045779?pr=43172

ci broken

**Comment by afc163** [comment]
> I noticed that there is an issue on caption missing right border. Same https://github.com/react-component/table/issues/999 exist in rc-table.

That should be fixed.

**Comment by sarpere** [comment]
I created a [pull request](https://github.com/react-component/table/pull/1001) for https://github.com/react-component/table/issues/999.

**Comment by afc163** [comment]
Ant Design table demo should be fixed too.

**Comment by sarpere** [comment]
Firstly,  [react-component/table](https://github.com/react-component/table) need to be deployed with lates updates. We solved that issue in the package. If it's not necessary to wait deployment, I can add styling to the style files.

**Comment by afc163** [comment]
antd didn't depend rc-table css file, so you have to resolve it in this repository.

**Pull request closed without merging**